reconcile/deadmanssnitch.py

```diff
@@ -0,0 +1,187 @@
+        return summary
+
+    def apply_diff(self, diff: dict[str, str]) -> None:
+        match diff["action"]:
+            case self.CREATE_SNITCH:
+                self.create_snitch(diff)
+            case self.DELETE_SNITCH:
+                self.deadmanssnitch_api.delete_snitch(diff["token"])
+            case self.UPDATE_VAULT:
+                self.vault_client.write({"path": f"{self.settings.snitches_path}/deadmanssnitch-{diff['cluster_name']}-url", "data": diff['snitch_url']})
+
+
+    def create_snitch(self, diff: dict[str, str]) -> None:
+        tags = ["app-sre"]
+        alert_email = [self.settings.alert_email]
+        payload = {
+            "name": f"prometheus.{diff['cluster_name']}.devshift.net",
+            "alert_type": "Heartbeat",
+            "interval": "15_minute",
+            "tags": tags,
+            "alert_email": alert_email,
+            "notes": self.settings.notes_link,
+        }
+        try:
+            snitch = self.deadmanssnitch_api.create_snitch(payload=payload)
+            self.vault_client.write({"path": f"{self.settings.snitches_path}/deadmanssnitch-{diff['cluster_name']}-url", "data": snitch.check_in_url})
+        except Exception as e:
+            logging.error(str(e))
+            return None
+
+
+class DeadMansSnitchIntegration(QontractReconcileIntegration[NoParams]):
+    """ Integration to automate deadmanssnitch snitch api during cluster dressup."""
+    @property
+    def name(self) -> str:
+        return QONTRACT_INTEGRATION
+
+    def __init__(self) -> None:
+        super().__init__(NoParams())
+        self.qontract_integration_version = make_semver(0, 1, 0)
+
+    def get_current_state(self, deadmanssnitch_api: DeadMansSnitchApi, clusters: ClusterV1, snitch_secret_path: str) -> list[dict[str, Any]]:
+        # current state includes for deadmanssnithch response and associated secret in vault
+        current_state: list[dict[str, Any]] = []
+        try:
+            snitches = deadmanssnitch_api.get_snitches(tags=["app-sre"])
+        except Exception as e:
+            logging.error(str(e))
+            return current_state
+        # create snitch_map only for  the desired clusters
+        for cluster in clusters:
+            for snitch in snitches:
+                if snitch.get_cluster_name() == cluster.name:
+                    snitch_map = snitch.dict(by_alias=True)
+                    snitch_map["cluster_name"] = cluster.name
+                    try:
+                        full_secret_path = {"path": snitch_secret_path, "field": f"deadmanssnitch-{snitch_map['cluster_name']}-url"}
+                        # match the exact url value
+                        snitch_map["vault_snitch_value"] = self.secret_reader.read(full_secret_path).strip()
+                    except SecretNotFound:
+                        # need to create a new vault secret for this record.
+                        snitch_map["vault_snitch_value"] = SECRET_NOT_FOUND
```

Comment:
@hemslo  we set a value only for a specfic form of exception which is SecretNotFound.. A None value will cause ambiguity like is it because of secret not found? or connection error? or something else.. Settings a value confirms that this is due to absence of secret and hence should be created

Reply:
if you make `vault_snitch_value` as a field, value as `str | None`, then it's clearer than always string. Or have another explicit field `valut_snitch_exist` as `bool`.